### PR TITLE
Upgrade to Kafka 0.9.0.1. This should fix #30

### LINKED ISSUE #30: Upgrade to Kafka 0.9.0.0
Kafka released a new version 0.9.0.0, can the puppet module be updated to be used with Kafka 0.9.0.0

### PR DESCRIPTION
**Pull request by dhoppe**
I could use some help regarding the upgrade to Kafka 0.9.0.1, because I am getting the following error messages: 

Kafka Mirror:
```
[2016-03-16 12:25:58,259] ERROR Exception when starting mirror maker. (kafka.tools.MirrorMaker$)
org.apache.kafka.common.config.ConfigException: Missing required configuration "bootstrap.servers" which has no default value.
	at org.apache.kafka.common.config.ConfigDef.parse(ConfigDef.java:148)
	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:49)
	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:56)
	at org.apache.kafka.clients.producer.ProducerConfig.<init>(ProducerConfig.java:315)
	at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:181)
	at kafka.tools.MirrorMaker$MirrorMakerProducer.<init>(MirrorMaker.scala:589)
	at kafka.tools.MirrorMaker$.main(MirrorMaker.scala:201)
	at kafka.tools.MirrorMaker.main(MirrorMaker.scala)
Exception in thread "main" java.lang.NullPointerException
	at kafka.tools.MirrorMaker$.main(MirrorMaker.scala:276)
	at kafka.tools.MirrorMaker.main(MirrorMaker.scala)
Exception in thread "MirrorMakerShutdownHook" java.lang.NullPointerException
	at kafka.tools.MirrorMaker$.cleanShutdown(MirrorMaker.scala:364)
	at kafka.tools.MirrorMaker$$anon$2.run(MirrorMaker.scala:187)
```

Kafka Producer:
```
java.lang.InstantiationException: kafka.tools.ConsoleProducer
	at java.lang.Class.newInstance(Class.java:368)
	at kafka.tools.ConsoleProducer$.main(ConsoleProducer.scala:39)
	at kafka.tools.ConsoleProducer.main(ConsoleProducer.scala)
```

### CODE REVIEW

**Comment by jyaworski** [comment]
LGTM.

**Comment by igalic** [comment]
do our acceptance tests actually run kafka to see if it comes up and works?

**Comment by dhoppe** [comment]
I just updated the acceptance tests, yesterday. But the Mirror and Producer service will quit with the error messages from above and I do not know why.

**Comment by dhoppe** [comment]
The latest commits will fix the errors regarding the Kafka mirror.

File: manifests/params.pp [hunk 1]
```diff
@@ -23,121 +22,269 @@
...
-    'auto.leader.rebalance.enable'                  => true,
-    'leader.imbalance.per.broker.percentage'        => '10',
-    'leader.imbalance.check.interval.seconds'       => '300',
-    'offset.metadata.max.bytes'                     => '1024',
+    'controller.socket.timeout.ms'                  => '30000',
+    'default.replication.factor'                    => '1',
+    'fetch.purgatory.purge.interval.requests'       => '1000',
+    'group.max.session.timeout.ms'                  => '30000',
+    'group.min.session.timeout.ms'                  => '6000',
+    'inter.broker.protocol.version'                 => '0.8.2.2',
```
Comment (igalic) [inline manifests/params.pp]: this is not the same as the program version?

File: manifests/params.pp [hunk 2]
```diff
@@ -23,121 +22,269 @@
...
+    'sasl.kerberos.kinit.cmd'                       => '/usr/bin/kinit',
+    'sasl.kerberos.min.time.before.relogin'         => '60000',
+    'sasl.kerberos.principal.to.local.rules'        => '[DEFAULT]',
+    'sasl.kerberos.service.name'                    => '',
+    'sasl.kerberos.ticket.renew.jitter'             => '0.05',
+    'sasl.kerberos.ticket.renew.window.factor'      => '0.8',
+    'security.inter.broker.protocol'                => 'PLAINTEXT',
+    'ssl.cipher.suites'                             => '',
+    'ssl.client.auth'                               => 'none',
+    'ssl.enabled.protocols'                         => '[TLSv1.2, TLSv1.1, TLSv1]',
```
Comment (igalic) [inline manifests/params.pp]: TLSv1? is that a good idea?

File: manifests/params.pp (hunk 1 above)
Comment (dhoppe) [inline manifests/params.pp]: http://kafka.apache.org/documentation.html#upgrade

You can not set this to the current versio,  until the whole cluster has been upgraded?

File: manifests/params.pp (hunk 2 above)
Comment (jyaworski) [inline manifests/params.pp]: I'd say default to TLSv1.2, and they can add more protocols if needed. That's what I did for puppetserver.

Comment (dhoppe) [inline manifests/params.pp]: http://kafka.apache.org/documentation.html#security_ssl

These are the default values, based on the current documentation and SSL is not enabled by default.

Comment (igalic) [inline manifests/params.pp]: ACK

**Comment by igalic** [comment]
:+1: from me: merge away when happiness levels are appropriate

**Pull request merged**